Whiskey/Functions/Install-WhiskeyDotNetTool.ps1

```diff
@@ -42,93 +43,114 @@ function Install-WhiskeyDotNetTool
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
-    if ($Version)
+    $WorkingDirectory = Resolve-Path -Path $WorkingDirectory
+    if (-not $WorkingDirectory)
     {
-        $sdkVersion = Resolve-WhiskeyDotNetSdkVersion -Version $Version
-        $installRoot = Join-Path -Path $InstallRoot -ChildPath '.dotnet'
-        return Install-WhiskeyDotNetSdk -InstallRoot $installRoot -Version $sdkVersion
+        return
     }
 
-    if ( (Get-Command -Name 'dotnet' -ErrorAction Ignore) )
+    Push-Location $WorkingDirectory
+    try
     {
-        # This command prints out the version of the .NET SDK that the "dotnet" command would use after taking into
-        # account the global.json file if it exists. Exit code 0 means a compatible SDK was found
-        $possibleVersion = dotnet --version 2>&1
-        if ( $LASTEXITCODE -eq 0 )
+        if ($Version)
         {
-            $msg = "[$($MyInvocation.MyCommand)] Using globally installed .NET SDK version $($possibleVersion)"
-            Write-WhiskeyVerbose -Message $msg
-            return 'dotnet'
+            $sdkVersion = Resolve-WhiskeyDotNetSdkVersion -Version $Version
+            $installRoot = Join-Path -Path $InstallRoot -ChildPath '.dotnet'
+            return Install-WhiskeyDotNetSdk -InstallRoot $installRoot -Version $sdkVersion
         }
-    }
-
-    $globalJsonPath = Join-Path -Path $WorkingDirectory -ChildPath 'global.json'
-    if ( -not (Test-Path -Path $globalJsonPath -PathType Leaf) )
-    {
-        $globalJsonPath = Join-Path -Path $InstallRoot -ChildPath 'global.json'
-    }
 
-
-
-    $sdkVersion = $null
-    if ( Test-Path -Path $globalJsonPath -PathType Leaf )
-    {
-        try
-        {
-            $globalJson = Get-Content -Path $globalJsonPath -Raw | ConvertFrom-Json
-        }
-        catch
+        if ((Get-Command -Name 'dotnet' -ErrorAction Ignore))
         {
-            Write-WhiskeyError -Message "global.json file ""$($globalJsonPath)"" contains invalid JSON."
-            return
+            # This command prints out the version of the .NET SDK that the "dotnet" command would use after taking into
+            # account the global.json file if it exists. Exit code 0 means a compatible SDK was found
+            $possibleVersion = Invoke-Command -ScriptBlock {
+                $ErrorActionPreference = 'Continue'
+                dotnet --version 2>$null
+            }
+            if ( $LASTEXITCODE -eq 0 )

```

Comment:
Hmm `$LASTEXITCODE` comes through `Invoke-Command -ScriptBlock { }`? I suppose it would if it's always set at the `$Global:` scope.

Reply:
I don't think `Invoke-Command` creates a new context. If there is an exiting variable in a parent scope, it use/sets it.